src/xbot/common/controls/sensors/MockXboxController.java

```diff
@@ -13,15 +16,62 @@ public MockXboxController(int port) {
     XYPair rightJoystickAxis;
     double leftTriggerAxis;
     double rightTriggerAxis;
+    private boolean xRightInverted = false;
+    private boolean yRightInverted = false;
+    private boolean xLeftInverted = false;
+    private boolean yLeftInverted = false;
+    
+    Map<Integer, Boolean> buttons = new HashMap<Integer, Boolean>();
+    
+    public void setRightX(double x) {
+        rightJoystickAxis.x = x;
+    }
+
+    public void setLeftX(double x) {
+        leftJoystickAxis.x = x;
+    }
+    
+    public void setRightY(double y) {
+        rightJoystickAxis.y = y;
+    }
+
+    public void setLeftY(double y) {
+        leftJoystickAxis.y = y;
+    }
+
+    public void pressButton(int button) {
+        buttons.put(button, true);
+    }
+
+    public void releaseButton(int button) {
+        buttons.put(button, false);
+    }
+
+    public XYPair getRightVector() {
+        return new XYPair(this.rightJoystickAxis.x * (xRightInverted ? -1 : 1), this.rightJoystickAxis.y
+                * (yRightInverted ? -1 : 1));
+    }
+
+    public XYPair getLeftVector() {
+        return new XYPair(this.leftJoystickAxis.x * (xLeftInverted ? -1 : 1), this.leftJoystickAxis.y
+                * (yLeftInverted ? -1 : 1));
+    }
     
-    boolean[] buttonValues = new boolean[10];
+    public boolean getRawButton(int button) {
+        return getButton(button);
+    }
+    
+    public boolean getButton(int button)
+    {
+        return buttons.getOrDefault(button, false);
+    }
     
     public void setButton(XboxButtons buttonName, boolean pressed) {
```

Comment:
If the `getButton` interface has one that takes an int and one that takes a button enum, the `setButton` interface should too. Then pick one of those two versions and call it everywhere else you need to set a button.